File: test/common/hash/Hash.java
public class Hash {
	public static long hash( byte[] data ) {
		int hash = 388650013;
		int scale = 388650179;
		int hardener  = 1176845762;
		int i = 0, n = data.length;
		while ( n > 0 ) {
			hash = ( hash * scale ) & 0xFFFFFFFF;
			hash = ( hash + data[ i++ ] ) & 0xFFFFFFFF;
			n--;
		}
		return ( ( hash ^ hardener ) & ( -1L >>> 32 ) );
	}

	public static void main( String[] args ) {
		if ( args.length != 1 ) {
			System.err.println( "Usage: java Hash [input string]" );
			return;
		}

		byte[] input = args[ 0 ].getBytes();

		System.out.println( Hash.hash( input ) );
	}
}

File: test/common/hash/hash.cc
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include "../../../common/hash/hash_func.hh"

int main( int argc, char **argv ) {
	if ( argc <= 1 ) {
		fprintf( stderr, "Usage: %s [input string]\n", argv[ 0 ] );
		return 1;
	}
	char *input = argv[ 1 ];
	printf( "%u\n", HashFunc::hash( input, ( unsigned int ) strlen( input ) ) );
	return 0;
}
